lib/diaspora_federation/entity.rb

```diff
@@ -319,8 +319,8 @@ def simple_node(doc, name, value)
       when :integer
         text.to_i if text =~ /^\d+$/
       when :boolean
-        return true if text =~ /(true|t|yes|y|1)$/i
-        false if text =~ /(false|f|no|n|0)$/i
+        return true if text =~ /^(true|t|yes|y|1)$/i
+        false if text =~ /^(false|f|no|n|0)$/i
```

Comment:
I wouldn't even discuss as much what makes sense or not on the protocol level, it should be a coding style rule to use `\A`/`\z`unless one explicitly wants the line-wise behavior.